packages/flutter/lib/src/cupertino/context_menu.dart

```diff
@@ -0,0 +1,1283 @@
+      return _ContextMenuLocation.center;
+    }
+
+    if (childRect.center.dx > center) {
+      return _ContextMenuLocation.right;
+    }
+
+    return _ContextMenuLocation.left;
+  }
+
+  // Push the new route and open the CupertinoContextMenu overlay.
+  void _openContextMenu() {
+    setState(() {
+      _childHidden = true;
+    });
+
+    _route = _ContextMenuRoute<void>(
+      actions: widget.actions,
+      barrierLabel: 'Dismiss',
+      filter: ui.ImageFilter.blur(
+        sigmaX: 5.0,
+        sigmaY: 5.0,
+      ),
+      contextMenuLocation: _contextMenuLocation,
+      previousChildRect: _decoyChildEndRect,
+      builder: (BuildContext context, Animation<double> animation) {
+        if (widget.previewBuilder == null) {
+          return widget.child;
+        }
+        return widget.previewBuilder(context, animation, widget.child);
+      },
+    );
+    Navigator.of(context, rootNavigator: true).push<void>(_route);
+    _route.animation.addStatusListener(_routeAnimationStatusListener);
+  }
+
+  void _onDecoyAnimationStatusChange(AnimationStatus animationStatus) {
+    switch (animationStatus) {
+      case AnimationStatus.dismissed:
+        if (_route == null) {
+          setState(() {
+            _childHidden = false;
+          });
+        }
+        _lastOverlayEntry?.remove();
+        _lastOverlayEntry = null;
+        break;
+
+      case AnimationStatus.completed:
+        setState(() {
+          _childHidden = true;
+        });
+        _openContextMenu();
+        // Keep the decoy on the screen for one extra frame. We have to do this
+        // because _ContextMenuRoute renders its first frame offscreen.
+        // Otherwise there would be a visible flash when nothing is rendered for
+        // one frame.
+        SchedulerBinding.instance.addPostFrameCallback((Duration _) {
+          _lastOverlayEntry?.remove();
+          _lastOverlayEntry = null;
+          _openController.reset();
+        });
+        break;
+
+      default:
+        return;
+    }
+  }
+
+  // Watch for when _ContextMenuRoute is closed and return to the state where
+  // the CupertinoContextMenu just behaves as a Container.
+  void _routeAnimationStatusListener(AnimationStatus status) {
+    if (status != AnimationStatus.dismissed) {
+      return;
+    }
+    setState(() {
+      _childHidden = false;
+    });
+    _route.animation.removeStatusListener(_routeAnimationStatusListener);
+    _route = null;
+  }
+
+  void _onTap() {
+    if (_openController.isAnimating && _openController.value < 0.5) {
+      _openController.reverse();
+    }
+  }
+
+  void _onTapCancel() {
+    if (_openController.isAnimating && _openController.value < 0.5) {
+      _openController.reverse();
+    }
+  }
+
+  void _onTapUp(TapUpDetails details) {
+    if (_openController.isAnimating && _openController.value < 0.5) {
+      _openController.reverse();
+    }
+  }
+
+  void _onTapDown(TapDownDetails details) {
```

Comment:
Hi, @justinmc, Why not trigger by `onLongPress` here?
![image](https://user-images.githubusercontent.com/61075224/99787112-49caa380-2b5a-11eb-903d-0c5fca1f54ac.png)

Reply:
If I remember correctly, the animation needs to start immediately on tap down.  I wait for `kLongPressTimeout` until I consider the context menu to be fully open.  If the tap is canceled before then, then it animates back and doesn't open.

The context menu is effectively opened on long press, not on tap, even though that's not the gesture I'm listening to.